### PR TITLE
notmuch: update to 0.21

### PR DESCRIPTION
**Pull request by diogoleal**

### CODE REVIEW

**Comment by dominikh** [comment]
@diogoleal Please fix the spelling mistake in the commit message. 
@xtraeme Could you take a look at http://sprunge.us/NYTJ ? It's the current travis build failure. The download fails with "Operation not permitted", and I cannot reproduce it locally. Is that a flaky issue in xbps-src or in travis? (Copied the log to sprunge so it doesn't get lost on the next push)

File: srcpkgs/notmuch/template [hunk 1]
```diff
@@ -69,3 +69,4 @@ notmuch-mutt_package() {
 		vman contrib/notmuch-mutt/notmuch-mutt.1
 	}
 }
+
```
Comment (dominikh) [inline srcpkgs/notmuch/template]: Please remove the extra newline.

**Comment by xtraeme** [comment]
@dominikh the travis issue with SSL certificates was fixed like a week ago. I'd think the void-packages git repo where this was submitted was outdated or not synced with origin/master?

**Comment by dominikh** [comment]
@xtraeme  Indeed. Sorry for the noise. @diogoleal, can you rebase on top of current master? Thanks (Also still the typo in the commit message. notmutch != notmuch)

**Comment by diogoleal** [comment]
@xtraeme same error. You want me to send you the output of the generation of the packages?

**Comment by Duncaen** [comment]
Try to rebase your changes onto master. `This branch is 1 commit ahead, 382 commits behind voidlinux:master.`

**Comment by diogoleal** [comment]
I made the base and still did not work tests.

**Comment by stpx** [comment]
I'm able to reproduce this.

```sh
$ curl https://notmuchmail.org/releases/notmuch-0.21.tar.gz
curl: (60) SSL certificate problem, verify that the CA cert is OK. Details:
error:14090086:SSL routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed
More details here: http://curl.haxx.se/docs/sslcerts.html

curl performs SSL certificate verification by default, using a "bundle"
 of Certificate Authority (CA) public keys (CA certs). If the default
 bundle file isn't adequate, you can specify an alternate file
 using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
 the bundle, the certificate verification probably failed due to a
 problem with the certificate (it might be expired, or the name might
 not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
 the -k (or --insecure) option.
```

**Comment by pullmoll** [comment]
Here's some more info:
```
jbu@box$ openssl s_client -connect notmuchmail.org:443 |tee logfile
depth=2 C = IL, O = StartCom Ltd., OU = Secure Digital Certificate Signing, CN = StartCom Certification Authority
verify error:num=19:self signed certificate in certificate chain
verify return:0
CONNECTED(00000003)
---
Certificate chain
 0 s:/C=US/CN=www.cworth.org/emailAddress=postmaster@cworth.org
   i:/C=IL/O=StartCom Ltd./OU=Secure Digital Certificate Signing/CN=StartCom Class 1 Primary Intermediate Server CA
 1 s:/C=IL/O=StartCom Ltd./OU=Secure Digital Certificate Signing/CN=StartCom Certification Authority
   i:/C=IL/O=StartCom Ltd./OU=Secure Digital Certificate Signing/CN=StartCom Certification Authority
 2 s:/C=IL/O=StartCom Ltd./OU=Secure Digital Certificate Signing/CN=StartCom Class 1 Primary Intermediate Server CA
   i:/C=IL/O=StartCom Ltd./OU=Secure Digital Certificate Signing/CN=StartCom Certification Authority
---
Server certificate
-----BEGIN CERTIFICATE-----
MIIHNzCCBh+gAwIBAgIDFEK+MA0GCSqGSIb3DQEBCwUAMIGMMQswCQYDVQQGEwJJ
TDEWMBQGA1UEChMNU3RhcnRDb20gTHRkLjErMCkGA1UECxMiU2VjdXJlIERpZ2l0
YWwgQ2VydGlmaWNhdGUgU2lnbmluZzE4MDYGA1UEAxMvU3RhcnRDb20gQ2xhc3Mg
MSBQcmltYXJ5IEludGVybWVkaWF0ZSBTZXJ2ZXIgQ0EwHhcNMTQxMTExMTAzOTU0
WhcNMTUxMTEyMDg1NzI3WjBMMQswCQYDVQQGEwJVUzEXMBUGA1UEAxMOd3d3LmN3
b3J0aC5vcmcxJDAiBgkqhkiG9w0BCQEWFXBvc3RtYXN0ZXJAY3dvcnRoLm9yZzCC
AiIwDQYJKoZIhvcNAQEBBQADggIPADCCAgoCggIBAL80g405eAI+bPVq2/C5Y29B
m75RV5L9Nan6WZSbZDG5GhiRGiqG4rTA0czddXzo9wzurVLQSZaiJD4geah3kAgx
ReE0sMPH/ut5Qsnf8kmUm0u5N39ky/PvHtPth7IYjVoHgsynEQJ3XYW5ulPY0XIs
IrcW8H0jo00A/GwgikWGyW8F/V7L4ZYFEKVFNKOznsuNYjXJnCRHv++IxqrCoqA+
1KPYG0StoGphF2zgXWvERmRd4SgbsWJToDKjLWp6z5qcNaItg+x5KSAxVjxKjmhI
UvX2pOjpP2kq3/6yDHVlb4vM9JzpPLohE6+H2JHgF2uIztpZkTM0wjFGUG4dUDhW
dkB2GmQlpYHBVCkZqBKBnHImW78R7OIJ29Cyq2Tmsxt8NGj3OPi5ChMJY1BN8ZFy
j+Hcpp/hH96nA8EbSA7ZszxFLI/B2++Zppifb224RF+oSXEUHV/m3AE75JnBDnsF
kkAKHJanRROvPTFzd275CAO/U5+dimfcRlgeaQaNo086sU32Xdf4e/Cro9zKDjCH
DGz7ujXnl9F/QK8GhWVWGH1cxO9X/JcSZERzjOXGnGIOT6uIyMxUk+/HOd6ugW1/
Y+ywOMbQho2W4s3ZvAcZ2LqaHoq+xN5E/7Ug8w24efrEQ73VXSdDRKW9eTSL25yF
Fl25Q+ew53g52Mlma255AgMBAAGjggLfMIIC2zAJBgNVHRMEAjAAMAsGA1UdDwQE
AwIDqDATBgNVHSUEDDAKBggrBgEFBQcDATAdBgNVHQ4EFgQUlYvhya0MWGk6MJF1
7SRyja8UK5IwHwYDVR0jBBgwFoAU60I00Jiwq5/0G2sI98xkLu8OLEUwJQYDVR0R
BB4wHIIOd3d3LmN3b3J0aC5vcmeCCmN3b3J0aC5vcmcwggFWBgNVHSAEggFNMIIB
STAIBgZngQwBAgEwggE7BgsrBgEEAYG1NwECAzCCASowLgYIKwYBBQUHAgEWImh0
dHA6Ly93d3cuc3RhcnRzc2wuY29tL3BvbGljeS5wZGYwgfcGCCsGAQUFBwICMIHq
MCcWIFN0YXJ0Q29tIENlcnRpZmljYXRpb24gQXV0aG9yaXR5MAMCAQEagb5UaGlz
IGNlcnRpZmljYXRlIHdhcyBpc3N1ZWQgYWNjb3JkaW5nIHRvIHRoZSBDbGFzcyAx
IFZhbGlkYXRpb24gcmVxdWlyZW1lbnRzIG9mIHRoZSBTdGFydENvbSBDQSBwb2xp
Y3ksIHJlbGlhbmNlIG9ubHkgZm9yIHRoZSBpbnRlbmRlZCBwdXJwb3NlIGluIGNv
bXBsaWFuY2Ugb2YgdGhlIHJlbHlpbmcgcGFydHkgb2JsaWdhdGlvbnMuMDUGA1Ud
HwQuMCwwKqAooCaGJGh0dHA6Ly9jcmwuc3RhcnRzc2wuY29tL2NydDEtY3JsLmNy
bDCBjgYIKwYBBQUHAQEEgYEwfzA5BggrBgEFBQcwAYYtaHR0cDovL29jc3Auc3Rh
cnRzc2wuY29tL3N1Yi9jbGFzczEvc2VydmVyL2NhMEIGCCsGAQUFBzAChjZodHRw
Oi8vYWlhLnN0YXJ0c3NsLmNvbS9jZXJ0cy9zdWIuY2xhc3MxLnNlcnZlci5jYS5j
cnQwIwYDVR0SBBwwGoYYaHR0cDovL3d3dy5zdGFydHNzbC5jb20vMA0GCSqGSIb3
DQEBCwUAA4IBAQBjem4I1Cf9czDnm6xDoT3Ga69nIYMgIbYRSLtTuUj3lTubBek0
W+uqX43bw4ukM8+Uxh0MRocAjEUq4kbjTho2MKhsEN+b0bQ9TlnLekpn0bDi2FiX
u/tNp0+IOQoiETSRyhFJPijbFUcrZMqfXaHZ6W18MChtdeIQv2dIdhlVAcUTQOj2
imQ7uIYU5D1ajsKWMKlVLUmMrAMrtX0ngRW/UI/YxfgOZ1J+WAN0yWtGbOwutNnH
ZjAF/rGdp+KsOC6mxCKaek2tvQlJqPBz+fX/bL9/blZDOJ03N6DPHWyJvbKGZ1Q6
iOtemBXUaLsIkSSemkqqepxC5Dl9aCTP1LBY
-----END CERTIFICATE-----
subject=/C=US/CN=www.cworth.org/emailAddress=postmaster@cworth.org
issuer=/C=IL/O=StartCom Ltd./OU=Secure Digital Certificate Signing/CN=StartCom Class 1 Primary Intermediate Server CA
---
No client certificate CA names sent
---
SSL handshake has read 6668 bytes and written 514 bytes
---
New, TLSv1/SSLv3, Cipher is ECDHE-RSA-AES256-GCM-SHA384
Server public key is 4096 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : ECDHE-RSA-AES256-GCM-SHA384
    Session-ID: 0DEC25D92E57D4CBE6C0D061150989696889925990BC5F5DCE7C14B8F9511536
    Session-ID-ctx: 
    Master-Key: A0EB45A6CCF442D678FF2785896C3B02C880988F44561695F5C363C93B5FE3DD55FD5EF89EB0C8C5B2B48D612D744B42
    TLS session ticket lifetime hint: 3600 (seconds)
    TLS session ticket:
    0000 - d8 d7 4a f8 06 01 38 c6-4e 5e 00 bb f6 36 e2 3c   ..J...8.N^...6.<
    0010 - dc d1 43 56 e7 63 99 46-c5 00 61 f7 01 48 99 8b   ..CV.c.F..a..H..
    0020 - 25 11 73 e4 3c d1 b8 4b-bf e9 c5 97 87 39 76 c1   %.s.<..K.....9v.
    0030 - 79 38 6c 6b cc 53 be 41-16 f6 54 e7 00 1a c8 42   y8lk.S.A..T....B
    0040 - d9 3b b1 1e c9 cc 6c 37-2a 8d ae 70 f5 a1 3a 06   .;....l7*..p..:.
    0050 - 48 6b 59 67 d4 ce e1 05-f2 cd c1 5b 71 43 9c 9f   HkYg.......[qC..
    0060 - 43 1d 25 b1 12 67 31 87-5b 30 82 86 00 9f dc 6e   C.%..g1.[0.....n
    0070 - c0 d3 f8 4d 4a b6 bc 84-32 27 79 ad dc 78 53 b0   ...MJ...2'y..xS.
    0080 - e2 e7 4e 54 73 0b 15 07-fa f4 cd 32 d9 a5 ce 48   ..NTs......2...H
    0090 - 44 63 be e5 ce 37 15 23-e8 6c ab 2d 58 86 d8 6f   Dc...7.#.l.-X..o
    00a0 - d7 80 fb 75 25 30 c4 d7-a9 97 9e 00 86 3c 9b c4   ...u%0.......<..
    00b0 - fd 6b 76 4f 0f 6b fa 9c-2e 90 e3 9a ce ee 2d 35   .kvO.k........-5
    00c0 - dc fb 5a 7f 90 a4 76 1d-fa 15 32 3a 23 03 65 e9   ..Z...v...2:#.e.
    00d0 - 45 ef 40 60 dd 30 03 64-3e ae 3a 8c 4a aa af 19   E.@`.0.d>.:.J...
    00e0 - 78 08 01 16 82 11 e5 0d-ec a0 07 12 2d dc ab b6   x...........-...
    00f0 - 4e 75 ce 05 89 55 f4 45-47 b3 98 37 e8 79 ec 4f   Nu...U.EG..7.y.O
    0100 - af d7 50 d3 40 bf 3b 4c-f4 ef fa 76 bb 63 c1 c3   ..P.@.;L...v.c..
    0110 - cc 2c a5 7b 85 51 1b 6c-1b 99 26 1d 5f ee e5 c1   .,.{.Q.l..&._...
    0120 - f1 05 6b 2a 5f e3 a2 e9-12 6a dd 78 ca c9 45 48   ..k*_....j.x..EH
    0130 - 58 a0 5d 44 0f 7e 63 d0-db 1f d0 95 29 5a a7 4a   X.]D.~c.....)Z.J
    0140 - c9 60 21 d9 e3 4b 92 32-b9 33 92 09 6b c7 2e da   .`!..K.2.3..k...
    0150 - ae 75 ae cc 00 3e 60 fa-79 a7 1b 40 6f 57 bf cc   .u...>`.y..@oW..
    0160 - 5f 22 04 a9 fb 3c 92 9c-db 8e 92 7e 7c 44 87 7b   _"...<.....~|D.{
    0170 - ab 06 1b 13 7a 80                                 ....z.

    Start Time: 1447285103
    Timeout   : 300 (sec)
    Verify return code: 19 (self signed certificate in certificate chain)
---
quit
Please log in for access to this system.closed
```

**Comment by xtraeme** [comment]
Looks like the server certificate is not valid (self-signed certificate).

**Comment by pullmoll** [comment]
Yep, the certificate is bogus in some way. I'd suggest to use a mirror for now like: `http://http.debian.net/debian/pool/main/n/notmuch/notmuch_0.21.orig.tar.gz`

**Comment by Duncaen** [comment]
I have a patch for cross compiling, this fixes at least the configuration phase, but it stops later with undefined references.
```
armv7l-linux-gnueabihf-gcc command-line-arguments.o debugger.o gmime-filter-reply.o hooks.o notmuch.o notmuch-compact.o notmuch-config.o notmuch-count.o notmuch-dump.o notmuch-insert.o notmuc
h-new.o notmuch-reply.o notmuch-restore.o notmuch-search.o notmuch-setup.o notmuch-show.o notmuch-tag.o notmuch-time.o sprinter-json.o sprinter-sexp.o sprinter-text.o query-string.o mime-node
.o crypto.o tag-util.o -specs=/void-packages/common/environment/configure/gccspecs/hardened-ld -Wl,-z,relro -Wl,-z,now -Wl,--as-needed -Wl,-z,relro     -L/usr/armv7l-linux-gnueabihf/usr/lib -
Lutil -lutil -Llib -lnotmuch -Wl,--as-needed -lgmime-2.6 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -Wl,-rpath,/usr/lib -ltalloc -lz -o notmuch-shared
command-line-arguments.o: In function `parse_option':
command-line-arguments.c:(.text+0x2a0): undefined reference to `_internal_error'
command-line-arguments.c:(.text+0x410): undefined reference to `_internal_error'
notmuch-dump.o: In function `notmuch_database_dump':
notmuch-dump.c:(.text+0x17c): undefined reference to `hex_encode'
notmuch-dump.c:(.text+0x20c): undefined reference to `make_boolean_term'
notmuch-reply.o: In function `create_reply_message':
notmuch-reply.c:(.text+0xcc4): undefined reference to `sanitize_string'
notmuch-reply.c:(.text+0xdc4): undefined reference to `xstrdup'
notmuch-restore.o: In function `notmuch_restore_command':
notmuch-restore.c:(.text+0x22c): undefined reference to `gz_getline'
notmuch-restore.c:(.text+0x340): undefined reference to `gz_getline'
notmuch-restore.c:(.text+0x3e0): undefined reference to `xregcomp'
notmuch-restore.c:(.text+0x434): undefined reference to `gz_error_string'
notmuch-restore.c:(.text+0x4c0): undefined reference to `xregexec'
notmuch-restore.c:(.text+0x4f0): undefined reference to `talloc_strndup_named_const'
notmuch-restore.c:(.text+0x518): undefined reference to `talloc_strndup_named_const'
notmuch-restore.c:(.text+0x588): undefined reference to `strtok_len'
notmuch-restore.c:(.text+0x608): undefined reference to `parse_boolean_term'
notmuch-restore.c:(.text+0x660): undefined reference to `gz_error_string'
notmuch-search.o: In function `notmuch_search_command':
notmuch-search.c:(.text+0xf5c): undefined reference to `sanitize_string'
notmuch-search.c:(.text+0xf6c): undefined reference to `sanitize_string'
notmuch-search.c:(.text+0x1290): undefined reference to `make_boolean_term'
notmuch-tag.o: In function `tag_query':
notmuch-tag.c:(.text+0x134): undefined reference to `make_boolean_term'
tag-util.o: In function `parse_tag_line':
tag-util.c:(.text+0x938): undefined reference to `hex_decode_inplace'
tag-util.c:(.text+0x96c): undefined reference to `strtok_len'
tag-util.c:(.text+0x9a4): undefined reference to `strtok_len'
collect2: error: ld returned 1 exit status
Makefile.local:306: recipe for target 'notmuch-shared' failed
```

**Pull request merged**